pyseir/cli.py

```diff
@@ -52,114 +56,190 @@ def _generate_whitelist() -> pd.DataFrame:
 
         # Aggregate the results created so far into one timeseries of metrics in a DataFrame
-        nola_infection_rate = pyseir.rt.patches.patch_aggregate_rt_results(need_patch)
+        nola_infection_rate = pyseir.rt.patches.patch_aggregate_rt_results(
+            infection_rate_map, population_map
+        )
 
-        patched: List[pd.DataFrame] = []
         for fips in need_patch:
-            fips_infection_rate = nola_infection_rate.copy()
-            fips_infection_rate.insert(0, CommonFields.FIPS, fips)
-            patched.append(fips_infection_rate)
-            # TODO(tom): Delete when no longer read
-            output_path = pipeline.Region.from_fips(fips).run_artifact_path_to_write(
-                pyseir.utils.RunArtifact.RT_INFERENCE_RESULT
+            this_fips_infection_rate = nola_infection_rate.copy()
+            this_fips_infection_rate.insert(0, CommonFields.FIPS, fips)
+            # Make a new SubStatePipeline object with the new infer_df
+            pipeline_map[fips] = dataclasses.replace(
+                pipeline_map[fips], infer_df=this_fips_infection_rate,
             )
-            fips_infection_rate.to_json(output_path)
-        results = pd.concat(patched + [unpatched])
-
-    return results
 
-
-def _states_region_list(state: Optional[str], default: List[str]) -> List[pipeline.Region]:
-    """Create a list of Region objects containing just state or default."""
-    if state:
-        return [pipeline.Region.from_state(state)]
-    else:
-        return [pipeline.Region.from_state(s) for s in default]
-
-
-def _state_only_pipeline(
-    region: pipeline.Region, output_interval_days=1, output_dir=None,
-) -> model_fitter.ModelFitter:
-    assert region.is_state()
-
-    infer_rt.run_rt(infer_rt.RegionalInput.from_region(region))
-    fitter = model_fitter.run_state(region)
-    ensembles_input = ensemble_runner.RegionalInput.from_model_fitter(fitter)
-    ensemble_runner.run_region(ensembles_input)
-    return fitter
+    return list(pipeline_map.values())
 
 
 def _build_all_for_states(
     states: List[str],
     output_interval_days=4,
     output_dir=None,
-    skip_whitelist=False,
     states_only=False,
-    fips: str = None,
+    fips: Optional[str] = None,
 ):
     # prepare data
     _cache_global_datasets()
 
     # do everything for just states in parallel
     with Pool(maxtasksperchild=1) as p:
-        states_only_func = partial(
-            _state_only_pipeline, output_interval_days=output_interval_days, output_dir=output_dir,
-        )
         states_regions = [pipeline.Region.from_state(s) for s in states]
-        state_fitters: List[model_fitter.ModelFitter] = p.map(states_only_func, states_regions)
-        state_fitter_map = {f.region: f for f in state_fitters}
+        state_pipelines: List[StatePipeline] = p.map(StatePipeline.run, states_regions)
+        state_fitter_map = {p.region: p.fitter for p in state_pipelines}
```

Comment:
I know they're technically not in the same scope, but might be worth changing the pool above to be named `pool` rather than `p` to disambiguate

Reply:
Done